scalableminds/webknossos
# Allow resolution restrictions for explorative annotations

- Adds the option to restrict resolutions for explorational volume (and hybrid) annotations
- This can be used to annotate large structures in downsampled mags without generating large amounts of mag1 data
- Replaces the drop-down in the dashboard by this modal:
![image](https://user-images.githubusercontent.com/1390035/128711441-5f4edf7f-e089-4b53-a92c-23b1680be05e.png)



### URL of deployed dev instance (used for testing):
- https://resolutionrestrictionsexplorational.webknossos.xyz

### TODO
 - [x] create modal in front-end to select the resolution restrictions when creating new annotations
  - [x] select annotation type
  - [x] select fallback layer
  - [x] select resolution restrictions with slider
  - [x] add explaining tooltips
  - [x] fix flow errors
  - [x] Crop resolution slider to fallback layer resolutions? Backend does this, does the UI need to mirror this?
  - [x] disable resolution restrictions if skeleton-only is selected
 - [x] relax other limits designed to limit data amount if not needed (#5067) (volumetracing_accessor.js)

### Possible Follow-Up TODOs 
  - [x] Disable resolution restrictions if only one resolution is present?
  - [ ] Add more tooltips explaining each each disabled state case for the elements?
  - [ ] Use slider UI also in task type creation?

### Steps to Test:
 - create annotation directly with the New Annotation button, resolutions of volume should mirror those of the fallback layer or (if there is none) of the union of the dataset layers
 - create annotation via the new modal (dotted menu in dashboard dataset list)
  -  with varying annotation type
  -  with varying fallback layer
  -  with varying resolution restrictions
  - for different datasets (segmentation layer present or not, different resolution combinations)
 - No unexpected behavior should crop up.

### Issues:
- fixes #5148
- fixes #5067

------
